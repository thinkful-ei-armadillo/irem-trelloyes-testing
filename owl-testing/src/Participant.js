import React from 'react';

function Participant(props){
    return (
        <div className="Participant">
            <img src={props.avatar} alt="headshot"/>
            <h3>{props.name}</h3>
            <p>{props.onStage}</p>
            <p>{props.id}</p>
        </div>
    );
}

export default Participant;