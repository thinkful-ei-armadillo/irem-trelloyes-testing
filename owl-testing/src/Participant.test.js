import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './Participant';
import renderer from 'react-test-renderer';

describe('<App />', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<Participant 
  title="Test Participant" 
      avatar ="https://ia.media-imdb.com/images/M/MV5BMTY5NzE3NzU3MF5BMl5BanBnXkFtZTgwMjg0NTQ5MDE@._V1_UX214_CR0,0,214,317_AL_.jpg"
      name ="Sally Salonga"
      onStage="false"
      inSession="true"
  />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Participant title="Test Participant" 
      avatar ="https://ia.media-imdb.com/images/M/MV5BMTY5NzE3NzU3MF5BMl5BanBnXkFtZTgwMjg0NTQ5MDE@._V1_UX214_CR0,0,214,317_AL_.jpg"
      name ="Sally Salonga"
      onStage="false"
      inSession="true" />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    })
})