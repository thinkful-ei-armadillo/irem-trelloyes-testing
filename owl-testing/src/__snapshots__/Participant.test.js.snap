// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the UI as expected 1`] = `
<div
  className="Participant"
>
  <img
    alt="headshot"
    src="https://ia.media-imdb.com/images/M/MV5BMTY5NzE3NzU3MF5BMl5BanBnXkFtZTgwMjg0NTQ5MDE@._V1_UX214_CR0,0,214,317_AL_.jpg"
  />
  <h3>
    Sally Salonga
  </h3>
  <p>
    false
  </p>
  <p>
    true
  </p>
</div>
`;
